### DISCUSSION TITLE
"App not installed" on upgrade

### DISCUSSION THREAD

**Discussion by woeisme** [KRTirtho/spotube#489] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

currently running Spotube 2.3.0 on Android 10. 
When opening Spotube it prompts to download latest version 2.7.1, after install get the message :

> App Not Installed

### Expected Behavior

App should upgrade normally. 

### Steps to reproduce

1. open app
2. follow upgrade dialogue to github latest release page
3. download upgrade package ( Spotube-android-all-arch.apk)
4. Download completes
5. continue dialogue (Do you want to install an update to this existing application? Your existing data will not be lost)
6. Install fails with message "App not installed."

### Operating System

Android 10

### Spotube version

2.7.1

### Installation source

GitHub Releases (Binary)

### Additional information

Also tried download link from webpage

**Comment by KRTirtho** [comment]
In 2.3.0 the signature for the app was different so while trying to upgrade android fails to match the signature and throws Error. Uninstall the old version and then install the new one
